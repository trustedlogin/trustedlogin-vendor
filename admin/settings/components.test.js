import {
  Input,
  Notice,
  Select,
  Form,
  FormTable,
  BigButton,
  Submit,
} from "./components";
import { render, fireEvent, cleanup } from "@testing-library/react";

describe("Input", () => {
  afterEach(cleanup);

  it("renders", () => {
    const onChange = jest.fn();
    const { container } = render(
      <FormTable>
        <Input
          label={"Label"}
          name={"name"}
          value={"value"}
          onChange={onChange}
        />
      </FormTable>
    );

    expect(container).toMatchSnapshot();
  });
  it("changes", () => {
    const onChange = jest.fn();
    const { getByLabelText } = render(
      <FormTable>
        <Input
          label={"Label"}
          name={"name"}
          value={"value"}
          onChange={onChange}
        />
      </FormTable>
    );
    fireEvent.change(getByLabelText("Label"), {
      target: { value: "New Value" },
    });
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith("New Value");
  });
});

describe("Select", () => {
  afterEach(cleanup);

  it("renders", () => {
    const onChange = jest.fn();
    const { container } = render(
      <FormTable>
        <Select
          label={"Label"}
          name={"name"}
          value={"value"}
          onChange={onChange}
          options={[
            { value: "one", label: "One" },
            { value: "two", label: "Two" },
          ]}
        />
      </FormTable>
    );
    expect(container).toMatchSnapshot();
  });
  it("changes", () => {
    const onChange = jest.fn();
    const { container } = render(
      <FormTable>
        <Select
          label={"Label"}
          name={"name"}
          value={"value"}
          onChange={onChange}
          options={[
            { value: "one", label: "One" },
            { value: "two", label: "Two" },
          ]}
        />
      </FormTable>
    );
  });
});

describe("BigButton", () => {
  afterEach(cleanup);

  it("renders", () => {
    const { container } = render(
      <BigButton onClick={jest.fn()}>Text</BigButton>
    );
    expect(container).toMatchSnapshot();
  });
  it("click", () => {});
});

describe("Form", () => {
  afterEach(cleanup);

  it("renders", () => {
    const { container } = render(
      <Form>
        <Submit name="s" disabled={false} value={"Save"} />
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders with title", () => {
    const { container } = render(
      <Form title="Spaces">
        <Submit name="s" disabled={false} value={"Save"} />
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
  it("submits", () => {
    const onSubmit = jest.fn();
    const { getByTitle } = render(
      <Form title="Spaces" onSubmit={onSubmit}>
        <Submit name="s" disabled={false} value={"Save"} />
      </Form>
    );
    fireEvent.submit(getByTitle("Spaces"));
  });
});

describe("FormTable", () => {
  afterEach(cleanup);

  it("renders", () => {
    const { container } = render(
      <Form>
        <FormTable>
          <tr>
            <td>1</td>
          </tr>
        </FormTable>
      </Form>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders with title", () => {
    const { container } = render(
      <Form>
        <FormTable title="One">
          <tr>
            <td>1</td>
          </tr>
        </FormTable>
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("Submit", () => {
  afterEach(cleanup);

  it("renders", () => {
    const { container } = render(<Submit value={"Save"} name="r" />);
    expect(container).toMatchSnapshot();
  });
});
