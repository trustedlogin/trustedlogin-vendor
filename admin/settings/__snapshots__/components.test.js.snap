// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigButton renders 1`] = `
<div>
  <p
    class="big-button"
  >
    <button
      class="button button-secondary button-hero"
    >
      Text
    </button>
  </p>
</div>
`;

exports[`Form renders 1`] = `
<div>
  <form>
    <p
      class="submit"
    >
      <input
        class="button button-secondary button-hero"
        id="s"
        name="s"
        type="submit"
        value="Save"
      />
    </p>
  </form>
</div>
`;

exports[`Form renders with title 1`] = `
<div>
  <form
    title="Spaces"
  >
    <h2
      class="title"
    >
      Spaces
    </h2>
    <p
      class="submit"
    >
      <input
        class="button button-secondary button-hero"
        id="s"
        name="s"
        type="submit"
        value="Save"
      />
    </p>
  </form>
</div>
`;

exports[`FormTable renders 1`] = `
<div>
  <form>
    <table
      class="form-table"
      role="presentation"
    >
      <tbody>
        <tr>
          <td>
            1
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;

exports[`FormTable renders with title 1`] = `
<div>
  <form>
    <h2
      class="title"
    >
      One
    </h2>
    <table
      class="form-table"
      role="presentation"
    >
      <tbody>
        <tr>
          <td>
            1
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;

exports[`Input renders 1`] = `
<div>
  <table
    class="form-table"
    role="presentation"
  >
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <label
            for="name"
          >
            Label
          </label>
        </th>
        <td>
          <input
            class="regular-text ltr"
            id="name"
            name="name"
            type="text"
            value="value"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Select renders 1`] = `
<div>
  <table
    class="form-table"
    role="presentation"
  >
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <label
            for="name"
          >
            Label
          </label>
        </th>
        <td
          data-children-count="1"
        >
          <select>
            <option
              class="level-0"
              value="one"
            >
              One
            </option>
            <option
              class="level-0"
              value="two"
            >
              Two
            </option>
          </select>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Submit renders 1`] = `
<div>
  <p
    class="submit"
  >
    <input
      class="button button-secondary button-hero"
      id="r"
      name="r"
      type="submit"
      value="Save"
    />
  </p>
</div>
`;
